AlumniConnect/views.py

```diff
@@ -163,21 +185,65 @@ def profileedit(request, id):
         return HttpResponseRedirect('/')
 
 
+@login_required
+def profile_completion(request):
+    profile = Profile.objects.get(roll_no=request.user.username)
+    if request.method == 'POST':
+        if profile.user_type == 'A':
+            form = Alumni_NewRegister(
+                request.POST, request.FILES, instance=profile)
+        else:
+            form = Student_NewRegister(
+                request.POST, request.FILES, instance=profile)
+        # print (request.POST)
+        if form.is_valid():
+            try:
+                first_name, last_name = request.POST['name'].split(' ', 1)
+            except:
+                first_name = request.POST['name']
+                last_name = ""
+            # print (form.cleaned_data.get('date_of_joining'))
+            profile = form.save(commit=False)
+            profile.reg_no = reg_no_gen(
+                profile.programme, profile.branch, profile.year_of_admission)
+            profile.country = request.POST['country']
+            profile.state = request.POST['state']
+            profile.city = request.POST['city']
+            profile.user.first_name = first_name
+            profile.user.last_name = last_name
+            request.user.first_name = first_name
+            request.user.last_name = last_name
+            request.user.is_active = False
+            profile.save()
```

Comment:
Instead of adding all this information manually, why not just do `form.save()` here?? You can use `reg_no_gen` first thing after `if request.method == 'POST'`, something like `request.POST['reg_no'] = reg_no_gen(...)`.